ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/Update.scala

```diff
@@ -259,4 +368,92 @@ object Update {
       override def definiteAnswer: Boolean = GrpcStatuses.isDefiniteAnswer(status)
     }
   }
+
+  implicit val `Update to LoggingValue`: ToLoggingValue[Update] = {
+    case update: ConfigurationChanged =>
+      ConfigurationChanged.`ConfigurationChanged to LoggingValue`.toLoggingValue(update)
+    case update: ConfigurationChangeRejected =>
+      ConfigurationChangeRejected.`ConfigurationChangeRejected to LoggingValue`.toLoggingValue(
+        update
+      )
+    case update: PartyAddedToParticipant =>
+      PartyAddedToParticipant.`PartyAddedToParticipant to LoggingValue`.toLoggingValue(update)
+    case update: PartyAllocationRejected =>
+      PartyAllocationRejected.`PartyAllocationRejected to LoggingValue`.toLoggingValue(update)
+    case update: PublicPackageUpload =>
+      PublicPackageUpload.`PublicPackageUpload to LoggingValue`.toLoggingValue(update)
+    case update: PublicPackageUploadRejected =>
+      PublicPackageUploadRejected.`PublicPackageUploadRejected to LoggingValue`.toLoggingValue(
+        update
+      )
+    case update: TransactionAccepted =>
+      TransactionAccepted.`TransactionAccepted to LoggingValue`.toLoggingValue(update)
+    case update: CommandRejected =>
+      CommandRejected.`CommandRejected to LoggingValue`.toLoggingValue(update)
+  }
+
+  private object Logging {
+    def recordTime(timestamp: Timestamp): LoggingEntry =
+      "recordTime" -> timestamp.toInstant
+
+    def submissionId(id: Ref.SubmissionId): LoggingEntry =
+      "submissionId" -> id
+
+    def submissionIdOpt(id: Option[Ref.SubmissionId]): LoggingEntry =
+      "submissionId" -> id
+
+    def participantId(id: Ref.ParticipantId): LoggingEntry =
+      "participantId" -> id
```

Comment:
I understand that we would like to keep some of `T => LoggingEntry` methods private to specific classes/packages because of required domain context that is known only in such class/package. However, we could at least consider moving methods for `Ref` types to a common object. Currently we have multiple definitions for several of `Ref` types, e.g. for `Ref.SubmissionId`.
What do you think?

Reply:
I’d be fine with moving them to `Ref` but doesn’t have to be in this PR.

Reply:
I'd be somewhat OK with this.

TBH I'm not sure these methods add too much but clearly others find value in them, so I guess there's something to it. I was tempted to inline them all.

That said, I'd prefer to find a way to derive these automatically from the type. Right now `SubmissionId` and some others are just type aliases to `LedgerString` though, so that wouldn't work.